### PR TITLE
firstaid for gh1393 and tagging-project733

### PR DESCRIPTION
**Pull request by u-fischer**
This improves firstaids for amsthm (https://github.com/latex3/tagging-project/issues/733) and (https://github.com/latex3/tagging-project/issues/736) and cleveref (#1393) 

- Ready to merge

## Checklist of required changes before merge will be approved
- [x] Test file(s) added
- [x] Version and date string updated in changed source files
- [x] Relevant `\changes` entries in source included
- [x] Relevant `changes.txt` updated
- [n/a] Rollback provided (if necessary)?
- [ ] `ltnewsX.tex` (and/or `latexchanges.tex`) updated

### CODE REVIEW

File: required/firstaid/latex2e-first-aid-for-external-files.dtx [hunk 1]
```diff
@@ -717,6 +717,34 @@
             {[\@tempb][\arabic{page}][\cref@result]\thepage}{}{}{}}}% <------- five
           \@esphack}%
           } 
+%    \end{macrocode}
+% \changes{v1.1g}{2024/10/16}{update cleveref data in label hook}
+% cleveref patches and redefines \cs{refstepcounter} so that a call updates its data.
+% This fails if like e.g. in \pkg{longtable} the counter is stepped with 
+% \cs{@kernel@refstepcounter}. We therefore move the data setup into the label hook.
+% As the hook is in a group we have to smuggle the data out of it.
+% \url{https://tex.stackexchange.com/a/722909/2388} and issue \#1393
```
Comment (FrankMittelbach) [inline required/firstaid/latex2e-first-aid-for-external-files.dtx:726]: or perhaps rethink the the  hook position behavior of "label"?

File: required/latex-lab/latex-lab-firstaid.dtx [hunk 2]
```diff
@@ -166,17 +167,27 @@
     \def\@begintheorem##1##2[##3]{%
 %    \end{macrocode}
 % We use the theorem instance.
+% \changes{v0.85h}{2024/10/16}{Added pre and post skips tagging/733}
 %    \begin{macrocode}
-      \UseInstance{blockenv}{theorem}{}
+      \UseInstance{blockenv}{theorem}{beginsep=\thm@preskip}
+%    \end{macrocode}
+% There is no working key to set the endskip, so we set the skip directly
```
Comment (FrankMittelbach) [inline required/latex-lab/latex-lab-firstaid.dtx:174]: that should be a temp situation: the end-skip key should work. So wouldn't it be better to adjust that in the block code to make it work first?

File: required/latex-lab/latex-lab-firstaid.dtx [hunk 3]
```diff
@@ -215,24 +229,17 @@
...
-      \tag_struct_begin:n{tag=Caption}  
       \thmname{##1}
       \thmnumber{
         \@ifnotempty{##1}{~}\@upn{##2}   
        }%   
-      \thmnote{\pdffakespace\space{\the\thm@notefont(##3)}}
-      %moved for easier tagging
-      \tag_mc_begin:n{}\the\thm@headpunct\tag_mc_end:  
-      \tag_struct_end:
+      \thmnote{\pdffakespace\space{\the\thm@notefont(##3)}}
```
Comment (FrankMittelbach) [inline required/latex-lab/latex-lab-firstaid.dtx:236]: Seeing this makes me wonder if one should hide the \pdffakespace\space combination ins something like \typesetspace

**Review by FrankMittelbach — APPROVED** [review]
only very superficially looked at but not much that sticks out

File: required/firstaid/latex2e-first-aid-for-external-files.dtx (hunk 1 above)
Comment (u-fischer) [reply required/firstaid/latex2e-first-aid-for-external-files.dtx:726]: I could change that as part of the PR.

File: required/latex-lab/latex-lab-firstaid.dtx (hunk 2 above)
Comment (u-fischer) [reply required/latex-lab/latex-lab-firstaid.dtx:174]: I looked at it but got a bit lost at the various (in part unused) skips you added there. I will open an proper issue that end-skip doesn't work and leave a note in the block code to adjust the firstaid once it works.

File: required/latex-lab/latex-lab-firstaid.dtx (hunk 3 above)
Comment (u-fischer) [reply required/latex-lab/latex-lab-firstaid.dtx:236]: I'm not sure where that could live currently. It is backend dependent and most code (for lualatex) is in tagpdf. Need some thoughts.

**Pull request merged**